route.go

```diff
@@ -638,3 +620,23 @@ func (mpp *multipathParser) AttributeDecoder() *netlink.AttributeDecoder {
 
 	return ad
 }
+
+// encodeIP is a helper for validating and encoding IPv4 and IPv6 addresses as
```

Comment:
For archival purpose:
I wonder how this would work on non-ip based networks, but I think it's safe to ignore for now and worry about that when we hit that snag.